src/function/Gaussian.cpp

```diff
@@ -11,6 +12,35 @@
 // Constructory Things //
 /////////////////////////
 
+// Constructory things
+Gaussian::Gaussian() : fFitter(this){
+    Initialise(std::vector<double>(1, 0), std::vector<double>(1, 1), "");
+}
+
+Gaussian::Gaussian(size_t nDims_, const std::string& name_): fFitter(this){
+    Initialise(std::vector<double>(nDims_, 0), std::vector<double>(nDims_, 1), name_);
+}// means = 0, stdDevs = 1
+
+Gaussian::Gaussian(double mean_, double stdDev_, const std::string& name_ ): fFitter(this){
+    Initialise(std::vector<double>(1, mean_), std::vector<double>(1, stdDev_), name_);
+}
+
+Gaussian::Gaussian(const std::vector<double>& mean_, 
+         const std::vector<double>& stdDev_, const std::string& name_ ): fFitter(this){
+    Initialise(mean_, stdDev_, name_);
+}
+
+Gaussian::Gaussian(const Gaussian& copy_): fFitter(this){
+    fMeans = copy_.fMeans;
+    fStdDevs = copy_.fStdDevs;
+    fCdfCutOff = copy_.fCdfCutOff;
+    fNDims = copy_.fNDims;
+    fName = std::string(copy_.fName+"_copy");
+    // THE ORDER HERE IS ALL IMPORTANT: INIT() AND GaussianFitter() MUST BE CALL AFTER THE MEANS AND STDDEVS ARE MADE!!!!!!!!!!
```

Comment:
This is a probable problem. The position of the  .init() call is important as the GaussianFitter  needs to know about the means and stddevs that are being copied. Any suggestions on how to fix this and make things more robust then i can definitely implement it them.